File: ruby/raindrops/raindrops.rb
class Raindrops
  # @return [String]
  def self.convert(i)
    running = ""
    running += "Pling" if i % 3 == 0
    running += "Plang" if i % 5 == 0
    running += "Plong" if i % 7 == 0
    running.empty? ? String(i) : running
  end
end

File: kotlin/raindrops/src/main/kotlin/Raindrops.kt
fun never(): Nothing = throw IllegalStateException()

object Raindrops {
    fun convert(n: Int) =
			listOf(3, 5, 7)
				.filter { n % it == 0 }
				.joinToString("") {
					when (it) {
						3 -> "Pling"
						5 -> "Plang"
						7 -> "Plong"
						else -> never()
					}
				}
				.ifEmpty { "$n" }
}
